folium/plugins/timeline.py

```diff
@@ -0,0 +1,289 @@
+          {% endfor %}
+
+          var {{ this.get_name() }} = L.timelineSliderControl(
+              {{ this.get_name() }}_options
+          );
+          {{ this.get_name() }}.addTo({{ this._parent.get_name() }});
+
+          {% for timeline in this.timelines %}
+              {{ this.get_name() }}.addTimelines({{ timeline.get_name() }});
+          {% endfor %}
+
+        {% endmacro %}
+    """
+    )
+
+    default_js = [
+        (
+            "timeline",
+            "https://cdn.jsdelivr.net/npm/leaflet.timeline@1.6.0/dist/leaflet.timeline.min.js",
+        ),
+        (
+            "moment",
+            "https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.18.1/moment.min.js",
+        ),
+    ]
+
+    def __init__(
+        self,
+        # arguments relevant to both interval and timestamp mode
+        auto_play: bool = True,
+        date_options: str = "YYYY-MM-DD HH:mm:ss",
+        start: Optional[Union[str, int, float]] = None,
+        end: Optional[Union[str, int, float]] = None,
+        enable_playback: bool = True,
+        enable_keyboard_controls: bool = False,
+        show_ticks: bool = True,
+        steps: int = 1000,
+        playback_duration: int = 10000,
+        **kwargs
+    ):
+        super().__init__()
+        self._name = "TimelineSlider"
+        self._getDisplayDateFormat = JsCode(
+            """
+            function(date) {
+                var newdate = new moment(date);
+                return newdate.format(\""""
+            + date_options
+            + """\");
+            }
+        """
+        )
+
+        kwargs["auto_play"] = auto_play
+        kwargs["start"] = start
+        kwargs["end"] = end
+        kwargs["enable_playback"] = enable_playback
+        kwargs["enable_keyboard_controls"] = enable_keyboard_controls
+        kwargs["show_ticks"] = show_ticks
+        kwargs["steps"] = steps
+        kwargs["duration"] = playback_duration
+        kwargs["format_output"] = self._getDisplayDateFormat
+
+        # extract JsCode objects
+        self.functions = {}
+        for key, value in list(kwargs.items()):
+            if isinstance(value, JsCode):
+                self.functions[camelize(key)] = value.js_code
+                kwargs.pop(key)
+
+        self.timelines: List[Timeline] = []
+        self.options = parse_options(**kwargs)
+
+    def add_timelines(self, *args):
+        """Add timelines to the control"""
+        self.timelines += args  # we do not check for duplicates
+        return self
+
+    def render(self, **kwargs):
+        assert isinstance(
+            self._parent, Map
+        ), "TimelineSlider can only be added to a Map object."
+        super().render(**kwargs)
+
+    def _get_self_bounds(self):
+        """
+        Computes the bounds of the object itself (not including it's children)
+        in the form [[lat_min, lon_min], [lat_max, lon_max]].
+
+        """
+        return get_bounds(self.data, lonlat=True)
```

Comment:
this method uses `self.data`, but that attribute doesn't exist. Maybe remove this method?

Reply:
Oops.